lib/engine/game/g_1849/step/buy_train.rb

```diff
@@ -1,12 +1,15 @@
 # frozen_string_literal: true
 
+require_relative '../../../step/base'
 require_relative '../../../step/buy_train'
 
 module Engine
   module Game
     module G1849
       module Step
         class BuyTrain < Engine::Step::BuyTrain
+          attr_accessor :e_token
```

Comment:
I didn't see step.e_token used anywhere. Is that correct? If so, then you don't need to expose it external to the step. @e_token suffices to use throughout the step.

Reply:
And actually seems to not be used at all, in which case can be removed.